sr_camera_pose_estimation/src/camera_transform_publisher.py

```diff
@@ -1,120 +1,140 @@
-                    rospy.logerr("Failed to find lens -> AR marker transform.")
-                    return
-                try:
-                    world_marker_trans = self.tfBuffer.lookup_transform(self.desired_camera_parent_frame,
-                                                                        static_marker_frame_name, rospy.Time())
-                except (tf2_ros.LookupException, tf2_ros.ConnectivityException, tf2_ros.ExtrapolationException):
-                    rospy.logerr("Failed to find world -> AR marker ({} -> {})transform.".format(
-                        self.desired_camera_parent_frame, static_marker_frame_name))
-                    return
-                transform = self.generate_world_camera_tf(lens_marker_trans, world_marker_trans)
-                self.publish_transform(transform)
+            if marker.id == self.marker_id:
+                self.on_new_pose(marker.pose.pose)
             else:
                 rospy.loginfo('Ignoring pose for marked id {}'.format(marker.id))
+    
+    def on_new_pose(self, pose):
+        self.counter += 1
+        self.publish_camera_pose(self.pose_averager.new_value(pose))
+    
+    def publish_camera_pose(self, lens_marker_pose):
+        try:
+            world_marker_trans = self.transform_buffer.lookup_transform(self.desired_camera_parent_frame,
+                                                                self.marker_static_tf_name, rospy.Time())
+        except (tf2_ros.LookupException, tf2_ros.ConnectivityException, tf2_ros.ExtrapolationException):
+            rospy.logerr("Failed to find world -> AR marker ({} -> {})transform.".format(
+                self.desired_camera_parent_frame, self.marker_static_tf_name))
+            return
+        self.transform = self.generate_world_camera_tf(lens_marker_pose, world_marker_trans)
+        self.publish_transform()
 
-    def generate_world_camera_tf(self, lens_marker_tf, world_marker_tf):
-        lens_marker_matrix = matrix_from_transform(lens_marker_tf.transform)
+    def generate_world_camera_tf(self, lens_marker_pose, world_marker_tf):
+        lens_marker_matrix = matrix_from_pose(lens_marker_pose)
         world_marker_matrix = matrix_from_transform(world_marker_tf.transform)
         world_lens_matrix = np.dot(world_marker_matrix, transformations.inverse_matrix(lens_marker_matrix))
         return transform_from_matrix(world_lens_matrix)
 
-    def publish_transform(self, transform):
+    def publish_transform(self):
         transform_stamped = TransformStamped()
         transform_stamped.header.stamp = rospy.get_rostime()
-        transform_stamped.transform = transform
+        transform_stamped.transform = self.transform
         transform_stamped.header.frame_id = self.desired_camera_parent_frame
         transform_stamped.child_frame_id = self.camera_root_frame
         self.broadcaster.sendTransform(transform_stamped)
 
+    def start_ar_track_alvar(self):
```

Comment:
So that the camera pose estimation node can start and stop the ar_track_alvar node. Part of the reason for making the camera pose estimation one-shot was to not have AR tracking eating CPU cycles constantly.